// rockSpace/lib/asteroids.js
exports.getRocks = function(){
    return space.getRocks();
};


canvas = { width:  700, height: 600 }

function Rock(){
    
};

Rock.prototype = {
    init: function(){
        this.width = 50;
        this.height = 50;                  
        this.r  = 0;
        this.speed = Math.random() * 1;
        this.x     = Math.random() * canvas.width;
        this.y     = Math.random() * canvas.height;
        this.dx    = (Math.random() * this.dir() * this.speed)+this.dir()*this.speed;
        this.dy    = (Math.random() * this.dir() * this.speed)+this.dir()*this.speed;
        this.rotationSpeed = Math.random() * 5 * this.dir();
    },
    
    update: function(){
        this.r += this.rotationSpeed;
        this.x += this.dx;
        this.y += this.dy;
        
        this.x = Math.round(this.x*100)/100;
        this.y = Math.round(this.y*100)/100;
        this.r = Math.round(this.r*10)/10;
        
        if(this.x < (-this.width) ){ this.x = canvas.width + this.width -1;  }
        if(this.x > canvas.width + this.width -1){ this.x = -this.width + 1;  }
        if(this.y < -this.height ){ this.y = canvas.height + this.height - 1; }
        if(this.y > canvas.height + this.height - 1){ this.y = -this.height + 1;  }
    },
    
    dir: function(){
        if(Math.random() > .51){ return 1 } else { return -1 }
    },
    
    simplePos: function(){ 
        return { 
            x: this.x, 
            y: this.y, 
            r: this.r 
        }; 
    }
};

space = {
    rockList: [],
    main: function(){
        this.createRocks(15);

    },
    createRocks: function(num){
        for( var amt = 0; amt < num; amt += 1){
            var newRock = new Rock();
            newRock.init();
            this.rockList.push(newRock);
        }
    },
    destroyRock: function(rockNum){
        
    
    },
    updateRockPos: function(){
        for( var rockNum = 0; rockNum < this.rockList.length; rockNum += 1){
            var tempRock = this.rockList[rockNum];
            tempRock.update();
        }
    },
    getRocks: function(){
        this.updateRockPos();
        var sendRocks = [];
        var rockNum = this.rockList.length;
        for(var cur = 0; cur < rockNum; cur += 1){
            var thisRock = this.rockList[cur].simplePos();
            sendRocks.push(thisRock);
        }
        return sendRocks;
    }
};
space.main();
